app/models/public_market.rb

```diff
@@ -2,14 +2,15 @@
 
 class PublicMarket < ApplicationRecord
   include FieldConstants
-  include FieldsValidation
 
   belongs_to :editor
+  has_and_belongs_to_many :market_types
+  has_and_belongs_to_many :market_attributes
```

Comment:
```suggestion
  has_and_belongs_to_many :attributes, class_name: 'MarketAttribute` do
```

si on peut se simplifier un peu de naming